develop-linux → SLua PV merge

marchcat opened the pull request:
Add Linux build support to SLua PV.

akleshchev committed:
Windows build fix

RyeMutt committed:
Remove dead LLPostProcess code with deprecated opengl usages

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Cherrypick 8de8daca601dc85e2b73687856f0a321016b4463 from archive/develop

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Restore llsys linux fixes from archive/develop 57d423745fd1d3d0ea6a0c69b869a20c27e27fc5

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Cherrypick 24854b4dd74edabf67efe533ef191553442a9bad from archive/develop

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix or supress various warnings under GCC builds

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix ODR violation during linux from ld

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix up linux feature table and various small build changes

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Disable viewer-manager in linux builds

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix build errors due to opengl headers and various defines

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix detection of AMD gpu on linux

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Cherrypick 42885c0d23d8efc47ad77e09bb327e4ec6a4fada from archive/develop

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Restore LLMaterialID to POD type

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix boost asio deprecation warnings

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix overflows inserting invalid values into stats map

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Cherrypick LLProcessor cleanup from 57d423745fd1d3d0ea6a0c69b869a20c27e27fc5

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Introduce support for GL_ATI_meminfo for getting gpu memory on linux

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Add debug option to force libvlc plugin on linux

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix numerous compiler warnings from GCC under Ubuntu 22.04

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Start to fix up GHA workflow for linux

Signed-off-by: Rye <rye@alchemyviewer.org>

dependabot[bot] committed:
Bump actions/stale from 6 to 10

Bumps [actions/stale](https://github.com/actions/stale) from 6 to 10.
- [Release notes](https://github.com/actions/stale/releases)
- [Changelog](https://github.com/actions/stale/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/stale/compare/v6...v10)

---
updated-dependencies:
- dependency-name: actions/stale
  dependency-version: '10'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

maxim-productengine committed:
#4411 WIP initial restoring of Legacy Search

akleshchev committed:
#4652 Improve switching of webrtc devices

Hecklezz committed:
Updated APR library and removed hotfix patch in APR cmake

marchcat committed:
Ensure final EOF in LLProcess test (#4877)

akleshchev committed:
#4876 Fix material not applying

1. Dupplicate call in llselectmgr can cause race conditions
2. lltooldragndrop wasn't updating server in one case

akleshchev committed:
p#494 Add search by agent ID to resident chooser

maxim-productengine committed:
#4411 WIP add Web tab to legacy Search

RyeMutt committed:
CMake changes for linux64 support including code from archive/develop
Restore changes for GLIB based dbus support from archive/develop

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix RelWithDebInfo build on mac universal

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix shader cache failing after clearing contents

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix low memory notification appearing incorrectly on non-windows platforms when changing graphics quality levels

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Cherry-pick gstreamer10 from 57d423745fd1d3d0ea6a0c69b869a20c27e27fc5

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix makefile and ninja generators invalidating secondlife-bin target every time cmake is run

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix transparent water shader being stuck as opaque when reset by feature table

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix crash observed during early startup during feature table resets on linux due to LLWorld simpleton not being initialized yet

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Restore linux CEF build support including changes from archive/develop
Fix various files missing linden_common

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
#4884 Crash at resetRegionCrossingTimer

akleshchev committed:
#4886 Crash on LLTeleportRequestViaLure::toOstream

RyeMutt committed:
Introduce global precompiled header to accelerate build

Signed-off-by: Rye <rye@alchemyviewer.org>

maxim-productengine committed:
#4411 fix additional search options - category and maturity

marchcat committed:
#4887 Fix access violation in LLInventoryModelBackgroundFetch::bulkFetchViaAis

RyeMutt committed:
Run tests using ctest after all targets are built to reduce build contention
Reuse test runner compilation unit to reduce extra source files compiled by 189
Only build macOS tests for build host architecture that will be running them to reduce wasted CI time

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Reintroduce llappearanceutility to viewer tree from archive/develop

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix shift by negative -1 in llflexibleobject

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix memcpy of 0 being reported as undefined behavior by ubsan

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix invalid cast warning from ubsan

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix leak of LLRender VBCache when reset/shutdown

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix broken mvp modelview hash check in syncMatrices

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix leak in materialmgr reported by asan/leaksan by using unique_ptr

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix uninitialized member variables reported by ubsan

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
Revert "Improve switching of webrtc devices"

Wrong approach. Might need to split workerDeployDevices into
separate recording and rendering variants.
This partiall reverts commit bb26aa3c2cb8ff961668cf0ad8180d3e9c57f941,
I left log lines.

RyeMutt committed:
Fix leak of boost signals in llmarketplacefunctions

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix leak of LLNotificationSeparator in LLFloaterNotificationsTabbed

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix GL debug logging on linux

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix shader warnings triggering GL errors under linux mesa drivers

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix assert failing debugGL run

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Add support for building with address, undefined, and thread sanitizers on linux

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix tracy memory tracking on linux to not crash mesa GL driver

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Introduce SDL based file picking support

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Introduce SDL3 based windowing backend support and dir picker

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Cleanup up linux wrapper script and fix desktop entry creation

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Modernize viewer alignment macros to use alignas and fix misaligned class issues reported by ubsan/asan

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Update openjpeg to 2.5.4-r1

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix broken locking for decomp data in model preview

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Modernize getAvailableMemoryKB for linux

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix deprecated opengl usage in debug display

Signed-off-by: Rye <rye@alchemyviewer.org>

maxim-productengine committed:
#4411 UI adjustments

maxim-productengine committed:
Fix indentation and trailing-whitespace

akleshchev committed:
#4879 An Emoji in a display name can affect text in chat

An emoji was spliting inlineview thus affecting displayed text.

maxim-productengine committed:
Mac build fix and clean up

RyeMutt committed:
Fix usage of deprecate opengl defines

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix deprecated opengl mipmap generation usage

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Reduce extra packages installed during autobuild configure

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix deprecated opengl draw commands

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Rework anisotropic filtering options into a dropdown with levels for better performance tuning of low end gpu

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Remove dead linux script

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix cef scaling on linux hidpi displays

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix broken mutex lock on skinned mesh map

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix non-static LLCachedControl

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Add early out for disk cache when app is shutting down to reduce shutdown time

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix multiple unicode file io handling issues with llofstream and llifstream

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Reduce texture cache start up time by reading header entries file in one call instead of many small ones

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix invalid calls to glLineWidth on some linux GL drivers

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix log to file not supporting multiline output

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
#3649 Incorrect camera aspec ratio

Everything else uses width/height ratio

RyeMutt committed:
Fix threaded GL contexts on linux backend

Enable core profile context support on linux

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
#3670 Add a sortable status column to the list of gestures

Geenz committed:
Merge branch 'develop-linux' into rye/linux64

akleshchev committed:
#1112 Add 1024x1024 size to Snapshot to Inventory

RyeMutt committed:
Add support for GLX and EGL to GL extension loader
Fix dullahan wayland support
Fix shader GL_ERROR under GL Core profile in Mesa

Signed-off-by: Rye <rye@alchemyviewer.org>

Geenz committed:
Merge pull request #4827 from RyeMutt/rye/linux64

Linux Support Part 1

RyeMutt committed:
Rework and modernize GL function/extension initialization for better wayland compatability

Signed-off-by: Rye <rye@alchemyviewer.org>

maxim-productengine committed:
Merge pull request #4900 from secondlife/maxim/2025.08-legacy-search

Restore Legacy Search

RyeMutt committed:
Remove dead linux crash logger

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix support for setting thread names on linux and macos

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix support for MSVC toolset v145(VS2026)

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Add basic linux build instructions

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix building with clang on non-darwin platforms

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Add support for tracy logging output

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Update open-libndof package with fixes for SDL3 joystick init

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix registration of secondlife protocol handler on linux

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
SL-3609 Fix camera POV jump when avatar crosses a region boundary

sFlycamPosition was using local coordinates instead of global ones thus
was not tracking global changes and was force setting local position to
an obsolete value.

Geenz committed:
Merge pull request #4915 from RyeMutt/rye/linux64-2

Linux64 Part 2

Geenz committed:
Merge branch 'develop' into develop-linux

maxim-productengine committed:
#4411 fix occurrences of fake BOOL

dependabot[bot] committed:
Bump actions/github-script from 7 to 8 (#4865)

Bumps [actions/github-script](https://github.com/actions/github-script) from 7 to 8.
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/v7...v8)

---
updated-dependencies:
- dependency-name: actions/github-script
  dependency-version: '8'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/labeler from 4 to 6 (#4866)

Bumps [actions/labeler](https://github.com/actions/labeler) from 4 to 6.
- [Release notes](https://github.com/actions/labeler/releases)
- [Commits](https://github.com/actions/labeler/compare/v4...v6)

---
updated-dependencies:
- dependency-name: actions/labeler
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Refactor labeler.yaml to use changed-files syntax

dependabot[bot] committed:
Bump actions/setup-python from 4 to 6 (#4922)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4 to 6.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4...v6)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Add bot* to allowlist in CLA workflow

akleshchev committed:
#4931 Fix name cache callbacks not having connections

RyeMutt committed:
Add GL constants needed for compatibility with physicsextensions_source package

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Improve fix for line width crashes under certain Mesa GL drivers

We default to and only use aliased lines.

Signed-off-by: Rye <rye@alchemyviewer.org>

Geenz committed:
Merge pull request #4934 from RyeMutt/rye/fixlnxbuild

Add GL constants needed for compatibility with physicsextensions_source package

Geenz committed:
Merge pull request #4935 from RyeMutt/rye/lnxfixlines

Improve fix for line width crashes under certain Mesa GL drivers

akleshchev committed:
#4946 Crash at setShaders

Fix wrong delimeter and log if file isn't present

RyeMutt committed:
Fix failure to generate build config information correctly under some cmake generators due to CMAKE_BUILD_TYPE only being utilized by non-multiconfig

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
Check what state pipe is in for a failed test (#4877)

maxim-productengine committed:
#4932 fix @mention being displayed twice if the name has a TM symbol

RyeMutt committed:
Fix alt cam mouse motion stuttering in variance to framerate under wayland

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix desktop entry lacking absolute path for Exec failing under KDE

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix GLTF files missing from model selection under SDL filepicking

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
#4955 Crash on itemChanged

maxim-productengine committed:
#4932 fix @mention being displayed twice if the name has a TM symbol

Geenz committed:
Remove cpuid feature bits verification in llprocessor.cpp (#4959)

Geenz committed:
Remove cpuid feature bits verification in llprocessor.cpp (#4959)

* Remove cpuid feature bits verification in llprocessor.cpp

maxim-productengine committed:
#4907 use shortcut to pan camera down

akleshchev committed:
#4671 Fix impractically long am/pm labels

RyeMutt committed:
Restore support for wavy avatar cloth(#4963)

Signed-off-by: Rye <rye@alchemyviewer.org>

Hecklezz committed:
Fix crash in updateImagesCreateTextures accessing potentially freed image pointer

Signed-off-by: Hecklezz <tj8@live.com.au>

Hecklezz committed:
Fix crash in updateDistance updating LOD on null mVObjp

Signed-off-by: Hecklezz <tj8@live.com.au>

Hecklezz committed:
Fix crash in handleHover accessing null parent

Signed-off-by: Hecklezz <tj8@live.com.au>

akleshchev committed:
#4939 Bugsplat update

to version 1.2.6

Hecklezz committed:
Follow-up null checks on mVObjp to prevent crashes

Signed-off-by: Hecklezz <tj8@live.com.au>

akleshchev committed:
#4980 Crashes when uploading a glTF model

Functions used in Image::prep aren't thread safe, pass them to main
thread and wait for a result

Geenz committed:
Disable the benchmark & occlusion on RDNA3.5 (#4999)

fmartian committed:
Fix error when adding a widget filename to the mFileNames stack

base_filename is already an absolute path that was resolved in ffindSkinnedFilenames() above but the oushFileName() method tries to do the same again, resulting in an invalid path (the skin directory path is prepended twice)

Hecklezz committed:
Fix the cache from being cleared every launch if NewCacheLocation is invalid

Signed-off-by: Hecklezz <tj8@live.com.au>

RyeMutt committed:
Remove leftover boost aligned_allocator as this is now handled by c++17 aligned allocations

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
#4948 Crash at webrtc's initializeConnection

akleshchev committed:
#4939 Bugsplat often confuses log files on mac, try to make it more reliable

akleshchev committed:
#4951 Update bugsplat symbol upload

akleshchev committed:
p#505 GLTF model upload fails to pick textures

maxim-productengine committed:
#5000 clear duplicate shortcut used to activate Sunrise

akleshchev committed:
#4991 Cant set cut off mask

maxim-productengine committed:
#4735 Remove the word "Viewer" from application shortcut

marchcat committed:
Merge branch 'develop-linux' into marchcat/slua-linux

# Conflicts:
#	indra/newview/llfeaturemanager.cpp
#	indra/newview/llstartup.cpp

akleshchev committed:
#4651 Improve session shutdown detection

akleshchev committed:
SL-9502 p#476 Fix visible hidden column

HaroldCindy commented:
Looks like websocketpp does not like compiling in C++20 mode on GCC https://github.com/zaphoyd/websocketpp/issues/991

akleshchev committed:
#5026 Upload To Folder feature not working

HaroldCindy commented:
Confirmed on Linux with GCC that the branch should build okay when https://github.com/secondlife/3p-websocketpp/pull/3 lands

akleshchev committed:
#5012 Switch 'chat' from untrusted, to click only

This command shouldn't be allowed from outside the app, but clicks should be valid

HaroldCindy committed:
Update websocketpp package to support Linux with C++20

HaroldCindy committed:
Try yet more patches to websocketpp

HaroldCindy commented:
Odd, seems like there's some issues with the build cache. There's intermittent windows errors with finding KDU and subsequent retries fail on hashing the autobuild.xml.

AiraYumi committed:
Prevent the creation of temporary objects that will be discarded.

Hecklezz committed:
Fixed LLWindowMacOSX::setCursorPosition scaling gl_pos when it shouldn't

Signed-off-by: Hecklezz <tj8@live.com.au>

HaroldCindy commented:
Should be okay, apparently AzureSignTool just updated, so that might be failing for other PRs as well.

FWIW multiple people have been using the linux build from this PR for a bit and things seem to be working fine

marchcat commented:
Thank you, @HaroldCindy!

Yeah, the build failure is unrelated to this PR. We'll likely fix it in `develop`.

akleshchev committed:
MacOS build fix

akleshchev committed:
#5035 Fix missed am/pm switch

RyeMutt committed:
Introduce menu option and key bind(alt+shift+t) to show/hide reflection probes in transparency debug view

marchcat committed:
Merge branch 'project/lua_editor' into marchcat/slua-linux

akleshchev committed:
#1963 Restore pelvis rotation debug settings

akleshchev committed:
#5046 Remove redundant updates in outfit list #1

akleshchev committed:
#5046 Fix accordion control's excessive rearranges #2

Since arrange is no longer part of LLInventoryItemsList::doIdle(),
reduced time limit.

akleshchev committed:
#5046 Optimize outfit's onRefreshComplete #3

RyeMutt committed:
Optimize post process rendering for gltf material preview

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix tabs in various source files introduced as part of legacy search

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Replace dependency on boost thread with std and change HttpService to use a similar thread shutdown mechanism to LLThread

akleshchev committed:
#1937 The "Don't show me in search" option is not visually updated

akleshchev committed:
#4322 Upload>Model button is sometimes disabled on MacOS

RyeMutt committed:
#5072 Use make_shared for more efficient ref counting and allocation

RyeMutt committed:
Fix occasional build failure observed when llphysicsextensionsos fails to build before secondlife-bin on macos with havok

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Replace boost noncopyable and undefined function usage with modern c++ deleted operator and constructor

RyeMutt committed:
#5078 Replace boost ptr_vector<nullable> with std::vector<std::unique_ptr>

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
#5078 Replace boost::hof::invocable with c++17 std::is_invocable

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
#5078 Remove dead llblowfish and llmail code plus related tests

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
#5078 Replace boost::function with std::function

* Replace boost::function usage with std::function for easier debugging and reduced compiler warnings
* Remove a few remaining instances of boost::noncopyable that were missed in tests

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
#5078 Replace boost::assign usage with modern c++ brace initialization

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix build failure introduced in boost::function cleanup from missing bind header

RyeMutt committed:
#5078 Remove dead experimental atmospherics code

Remove dead experimental atmospherics code left over from EEP development

RyeMutt committed:
#5083 Fallback to OS open handler in external editor when no editor is explicitly set (#5077)

* Fallback to OS open handler in external editor when no editor is explicitly set

Signed-off-by: Rye <rye@alchemyviewer.org>

* Apply suggestion to use SystemRoot for windows OS root path in open handler

Co-authored-by: Ansariel Hiller <Ansariel@users.noreply.github.com>

---------

Signed-off-by: Rye <rye@alchemyviewer.org>
Co-authored-by: Ansariel Hiller <Ansariel@users.noreply.github.com>

akleshchev committed:
#5078 MacOS Build fix

RyeMutt committed:
Fix alpha debug overlay rendering in probes and breaking scene lighting (#5058)

Signed-off-by: Rye <rye@alchemyviewer.org>

akleshchev committed:
#5084 Ressurect Watchdog

akleshchev committed:
#5086 Fix 'Copy SLURL' generating obsolete http link

use https

akleshchev committed:
#5084 Better labeling of uncaught events, freezes, crashes

At least while watchdog is enabled it's more fair to label these as unknowns

RyeMutt committed:
Fix invalid symbol in gstreamer 1.0 plugin crashing (#5059)

Signed-off-by: Rye <rye@alchemyviewer.org>

Geenz committed:
Merge pull request #4853 from secondlife/release/2025.08

Release/2025.08

Geenz committed:
Merge pull request #5094 from secondlife/main

2025.08 -> Develop

marchcat committed:
Merge branch 'develop-linux' into marchcat/slua-linux

marchcat committed:
Merge branch 'release/2026.01' into marchcat/devlinux-2601

# Conflicts:
#	indra/llcommon/llprocessor.cpp
#	indra/llcorehttp/_refcounted.h
#	indra/llcorehttp/_thread.h
#	indra/newview/llfeaturemanager.cpp
#	indra/newview/llviewershadermgr.cpp

marchcat committed:
LLTextBase build fix

marchcat committed:
Merge pull request #5101 from secondlife/marchcat/devlinux-2601

Merge 26.01 into develop-linux

marchcat committed:
Merge branch 'develop-linux' into marchcat/slua-linux

# Conflicts:
#	indra/newview/llfeaturemanager.cpp

marchcat committed:
Revert "Windows build fix"

This reverts commit 8c292db722d01fcaef95752f93843e8533b05710.

marchcat committed:
Proper Windows build fix (#5105)

marchcat committed:
Merge branch 'develop-linux' into marchcat/slua-linux
